central/cloudproviders/gcp/singleton.go

```diff
@@ -16,6 +17,9 @@ var (
 
 // Singleton returns an instance of the GCP cloud credentials manager.
 func Singleton() auth.STSClientManager {
+	if !features.CloudCredentials.Enabled() {
+		return nil
+	}
```

Comment:
Hmm, introducing this _without_ checking for nil within the `StorageManager()` function on the manager implementation leads to a nil pointer panic with the current usage of the code.

That's why we typically avoid doing the whole "return nil if feature flag is not set" _outside_ of the singleton.

Reply:
Added a check for the feature flag in the integration. We can remove this check again with the feature flag.